google-cloud-translate/lib/google/cloud/translate.rb

```diff
@@ -106,34 +108,40 @@ module Translate
       #   translation.text #=> "Salve mundi!"
       #
       def self.new project_id: nil, credentials: nil, key: nil, scope: nil,
-                   retries: nil, timeout: nil, project: nil, keyfile: nil
+                   retries: nil, timeout: nil, endpoint: nil, project: nil,
+                   keyfile: nil
         project_id ||= (project || default_project_id)
         key        ||= configure.key
+        retries    ||= configure.retries
+        timeout    ||= configure.timeout
+        endpoint   ||= configure.endpoint
 
         if key
           return Google::Cloud::Translate::Api.new(
             Google::Cloud::Translate::Service.new(
-              project_id.to_s, nil, retries: retries, timeout: timeout, key: key
+              project_id.to_s, nil,
+              retries: retries, timeout: timeout, key: key, host: endpoint
             )
           )
         end
 
         scope       ||= configure.scope
-        retries     ||= configure.retries
-        timeout     ||= configure.timeout
         credentials ||= keyfile || default_credentials(scope: scope)
 
         unless credentials.is_a? Google::Auth::Credentials
           credentials = Translate::Credentials.new credentials, scope: scope
         end
 
-        project_id ||= credentials.project_id if credentials.respond_to? :project_id
-        project_id = project_id.to_s # Always cast to a string
+        # project_id ||= credentials&.project_id
+        # project_id ||= credentials.project_id if credentials.respond_to? :project_id
+        # project_id = project_id.to_s # Always cast to a string
```

Comment:
Why are these lines commented out? Should they be removed?

Reply:
Yes. Removed.